RecoEgamma/PhotonIdentification/plugins/PhotonMVAEstimator.cc

```diff
@@ -1,34 +1,41 @@
 #include "RecoEgamma/PhotonIdentification/interface/PhotonMVAEstimator.h"
-
-PhotonMVAEstimator::PhotonMVAEstimator(const edm::ParameterSet& conf):
-  AnyMVAEstimatorRun2Base(conf),
-  name_(conf.getParameter<std::string>("mvaName")),
-  tag_(conf.getParameter<std::string>("mvaTag")),
-  methodName_("BDTG method"),
-  mvaVarMngr_(conf.getParameter<std::string>("variableDefinition")),
-  ebeeSplit_ (conf.getParameter<double> ("ebeeSplit")),
-  debug_(conf.getUntrackedParameter<bool>("debug", false))
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+PhotonMVAEstimator::PhotonMVAEstimator(const edm::ParameterSet& conf)
+  : AnyMVAEstimatorRun2Base(conf)
+  , mvaVarMngr_ (conf.getParameter<std::string>("variableDefinition"))
+  // So far only used in Run2Spring16NonTrigV1
+  , effectiveAreas_        (conf.exists("effAreasConfigFile") ?
```

Comment:
::exists method is rather an exception left for cases for rather old and complex pre-7X software.
Define a ::fillDescriptions method https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp

Reply:
But this is just a class and not a plugin. In this case, I can't implement a `::fillDescriptions` method, right? And the plugin which uses this class should not need to know which specific parameters this MVA expects I think.